--- conccgi.py.orig     2024-06-11 17:10:53.415128157 +0200
+++ conccgi.py  2024-06-11 16:53:30.741859415 +0200
@@ -1,4 +1,5 @@
 # Copyright (c) 2003-2020  Pavel Rychly, Vojtech Kovar, Milos Jakubicek, Milos Husak, Vit Baisa
+import unicodedata

 from CGIPublisher import correct_types
 from usercgi import UserCGI
@@ -17,6 +18,7 @@
 from annotlib import Annotation
 from conclib import strkwiclines

+UNICODE_NORMALIZED_FORM = 'NFD'

 def onelevelcrit (prefix, attr, ctx, pos, fcode, icase, bward='', empty=''):
     fromcode = {'lc': '<0', 'rc': '>0', 'kl': '<0', 'kr': '>0'}
@@ -904,7 +906,7 @@

     def _compile_query(self, qtype=None, cname=''):
         if not self.is_err_corpus():
-            return self._compile_basic_query(qtype, cname=cname)
+            return unicodedata.normalize(UNICODE_NORMALIZED_FORM, self._compile_basic_query(qtype, cname=cname))
         err_code = getattr(self, 'cup_err_code', '')
         err = getattr(self, 'cup_err', '')
         corr = getattr(self, 'cup_corr', '')
@@ -915,7 +917,7 @@
                 cql = cql.replace('][', '] (<corr/>)? [')
                 cql = cql.replace('](', '] (<corr/>)? (')
                 cql = cql.replace('] [', '] (<corr/>)? [')
-            return cql
+            return unicodedata.normalize(UNICODE_NORMALIZED_FORM, cql)
         # compute error query
         # Achtung: Top error lists (method freq) depends on this -- do check it
         # when changing the below
@@ -937,12 +939,12 @@
         fullstruct = '(%s%s)' % (err_query, corr_query)
         if self.cup_hl == 'e' or (self.cup_hl == 'q' and err_restr
                                                      and not corr_restr):
-            return '%s within %s' % (err_query, fullstruct)
+            return unicodedata.normalize(UNICODE_NORMALIZED_FORM, '%s within %s' % (err_query, fullstruct))
         elif self.cup_hl == 'c' or (self.cup_hl == 'q' and corr_restr
                                                        and not err_restr):
-            return '%s within %s' % (corr_query, fullstruct)
+            return unicodedata.normalize(UNICODE_NORMALIZED_FORM, '%s within %s' % (corr_query, fullstruct))
         else: # highlight both
-            return fullstruct
+            return unicodedata.normalize(UNICODE_NORMALIZED_FORM, fullstruct)

     def query (self, qtype='cql'):
